api/src/main/java/org/openmrs/api/context/OpenmrsApplicationContextConfig.java

```diff
@@ -0,0 +1,208 @@
+import org.openmrs.util.LocaleUtility;
+import org.openmrs.util.LocationUtility;
+import org.openmrs.util.TestTypeFilter;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.DependsOn;
+import org.springframework.context.annotation.FilterType;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.Resource;
+
+@Configuration
+@ComponentScan(
+    basePackages = "org.openmrs",
+    includeFilters = {
+        @ComponentScan.Filter(type = FilterType.ANNOTATION, classes = Handler.class),
+        @ComponentScan.Filter(type = FilterType.CUSTOM, classes = OpenmrsProfileIncludeFilter.class)
+    },
+    excludeFilters = {
+        @ComponentScan.Filter(type = FilterType.CUSTOM, classes = TestTypeFilter.class),
+        @ComponentScan.Filter(type = FilterType.CUSTOM, classes = OpenmrsProfileExcludeFilter.class)
+    }
+)
+public class OpenmrsApplicationContextConfig {
+
+    @Bean
+    public EventListeners clearOpenmrsEventListeners() {
+        EventListeners eventListeners = new EventListeners();
+        eventListeners.setGlobalPropertyListenersToEmpty(false);
+        return eventListeners;
+    }
+
+    @Bean
+    public LocaleUtility localeUtility() {
+        return new LocaleUtility();
+    }
+
+    @Bean
+    public LocationUtility locationUtility() {
+        return new LocationUtility();
+    }
+
+    @Bean
+    public ConfigUtil configUtilGlobalPropertyListener() {
+        return new ConfigUtil();
+    }
+
+    @Bean
+    public PersonNameGlobalPropertyListener personNameGlobalPropertyListener() {
+        return new PersonNameGlobalPropertyListener();
+    }
+
+    @Bean
+    public LoggingConfigurationGlobalPropertyListener loggingConfigurationGlobalPropertyListener() {
+        return new LoggingConfigurationGlobalPropertyListener();
+    }
+
+    @Bean
+    @DependsOn("clearOpenmrsEventListeners")
+    public EventListeners openmrsEventListeners(LocaleUtility localeUtility, LocationUtility locationUtility,
+            ConfigUtil configUtilGlobalPropertyListener,
+            PersonNameGlobalPropertyListener personNameGlobalPropertyListener,
+            LoggingConfigurationGlobalPropertyListener loggingConfigurationGlobalPropertyListener,
+            GlobalLocaleList globalLocaleList, AdministrationServiceImpl adminService,
+            OrderServiceImpl orderService) {
+        EventListeners eventListeners = new EventListeners();
+        List<GlobalPropertyListener> globalPropertyListeners = List.of(localeUtility, locationUtility,
+                configUtilGlobalPropertyListener, personNameGlobalPropertyListener,
+                loggingConfigurationGlobalPropertyListener, globalLocaleList, adminService, orderService);
+        eventListeners.setGlobalPropertyListeners(globalPropertyListeners);
+        return eventListeners;
```

Comment:
It's a very strange construct, but I'm fine with this for now. We'll refactor in a separate issue.